cmd/openshift-install/gather.go

```diff
@@ -132,38 +137,53 @@ func runGatherBootstrapCmd(directory string) (string, error) {
 		}
 
 		if bootstrap == "" || len(masters) == 0 {
-			return "", errors.New("bootstrap host address and at least one control plane host address must be provided")
+			return "", "", errors.New("bootstrap host address and at least one control plane host address must be provided")
 		}
 	} else if bootstrap == "" || len(masters) == 0 {
-		return "", errors.New("must provide both bootstrap host address and at least one control plane host address when providing one")
+		return "", "", errors.New("must provide both bootstrap host address and at least one control plane host address when providing one")
 	}
 
 	return gatherBootstrap(bootstrap, port, masters, directory)
 }
 
-func gatherBootstrap(bootstrap string, port int, masters []string, directory string) (string, error) {
+func gatherBootstrap(bootstrap string, port int, masters []string, directory string) (string, string, error) {
+	gatherID := time.Now().Format("20060102150405")
+
+	serialLogBundle := filepath.Join(directory, fmt.Sprintf("serial-log-bundle-%s.tar.gz", gatherID))
+	serialLogBundlePath, err := filepath.Abs(serialLogBundle)
+
+	logrus.Info("Pulling platform specific console logs from the bootstrap machine")
+	gather, err := gather.New(logrus.StandardLogger(), serialLogBundle, bootstrap, masters, directory)
+	if err != nil {
+		return "", "", errors.Wrap(err, "Failed to create gather object")
+	}
+	err = gather.Run()
+	if err != nil {
+		logrus.Infof("%s: Failed to gather platform specific logs", err.Error())
+	}
+
 	logrus.Info("Pulling debug logs from the bootstrap machine")
 	client, err := ssh.NewClient("core", net.JoinHostPort(bootstrap, strconv.Itoa(port)), gatherBootstrapOpts.sshKeys)
 	if err != nil {
 		if errors.Is(err, syscall.ECONNREFUSED) || errors.Is(err, syscall.ETIMEDOUT) {
 			return "", errors.Wrap(err, "failed to connect to the bootstrap machine")
 		}
-		return "", errors.Wrap(err, "failed to create SSH client")
+		return "", "", errors.Wrap(err, "failed to create SSH client")
 	}
 
-	gatherID := time.Now().Format("20060102150405")
 	if err := ssh.Run(client, fmt.Sprintf("/usr/local/bin/installer-gather.sh --id %s %s", gatherID, strings.Join(masters, " "))); err != nil {
-		return "", errors.Wrap(err, "failed to run remote command")
+		return "", "", errors.Wrap(err, "failed to run remote command")
 	}
 	file := filepath.Join(directory, fmt.Sprintf("log-bundle-%s.tar.gz", gatherID))
 	if err := ssh.PullFileTo(client, fmt.Sprintf("/home/core/log-bundle-%s.tar.gz", gatherID), file); err != nil {
-		return "", errors.Wrap(err, "failed to pull log file from remote")
+		return "", "", errors.Wrap(err, "failed to pull log file from remote")
 	}
-	path, err := filepath.Abs(file)
+	logBundlePath, err := filepath.Abs(file)
 	if err != nil {
-		return "", errors.Wrap(err, "failed to stat log file")
+		return "", "", errors.Wrap(err, "failed to stat log file")
 	}
-	return path, nil
+
+	return serialLogBundlePath, logBundlePath, nil
```

Comment:
We should try to have the serial console logs added to the bootstrap log bundle (rather than two separate bundles).

Reply:
Fixed.